### ISSUE TITLE
Should we forbid converting str to numeric values in Column.set()?

### ISSUE THREAD

**Issue by hagenw** [audeering/audformat#311]
When using `audformat.Column.set()` we only check if the provided value(s) match a corresponding scheme if the scheme has labels or a minimum and or maximum.

Let's start with a database containing an integer scheme:

```python
import audformat

db = audformat.Database('db')
db.schemes['scheme'] = audformat.Scheme('int')
db['table'] = audformat.Table(audformat.filewise_index('f1'))
db['table']['column'] = audformat.Column(scheme_id='scheme')
```

If we assign a string value it is converted to integer automatically:

```python
>>> db['table']['column'].set('1')
>>> db['table']['column'].get()
file
f1    1
Name: column, dtype: Int64
```

But if a user provides a string that cannot be converted to the requested numeric type she will get an error message that might be not that helpful:

```python
>>> db['table']['column'].set('a1')
...
ValueError: invalid literal for int() with base 10: 'a1'
```

There are two possible options to enhance this:
* forbid assigning strings to numeric schemes
* catch the ValueError and raise another scheme specific ValueError stating that the provided value cannot be converted to the requested dtype

/cc @hesamsagha

**Comment by frankenjoe** [comment]
> catch the ValueError and raise another scheme specific ValueError stating that the provided value cannot be converted to the requested dtype

Seems like the easier solution since there might be other implicit conversions we are not aware of.

**Comment by hagenw** [comment]
Yes, I would also be in favor of raising an error with a better error message.

**Comment by hagenw** [comment]
On the other hand catching the error might also hide the underlying problem, e.g. https://github.com/audeering/audformat/issues/364 would have been very hard to understand by a user if we would have catched the `ValueError` and replaced it with a custom message stating that the user cannot provide dates to a column with a date scheme ;)
